contracts/exchange_contract/src/main.sw

```diff
@@ -205,36 +177,36 @@ impl Exchange for Contract {
             // if token ratio is correct, proceed with liquidity operation
             // otherwise, return current user balances in contract
             if (current_token_amount >= token_amount) {
-              // Add fund to the reservers
-              add_reserve(TOKEN_ID, token_amount);
-              add_reserve(ETH_ID, current_eth_amount);
-              // Mint LP token
-              mint(liquidity_minted);
-              storage.lp_token_supply = total_liquidity + liquidity_minted;
-
-              transfer_to_output(liquidity_minted, contract_id(), sender);
-
-              // If user sent more than the correct ratio, we deposit back the extra tokens
-              let token_extra = current_token_amount - token_amount;
-              if (token_extra > 0) {
-                transfer_to_output(token_extra, ~ContractId::from(TOKEN_ID), sender);
-              }
-
-              minted = liquidity_minted;
+                // Add fund to the reserves
+                storage.token_reserve = storage.token_reserve + token_amount;
+                storage.eth_reserve = storage.eth_reserve + current_eth_amount;
+                // Mint LP token
+                mint(liquidity_minted);
+                storage.lp_token_supply = total_liquidity + liquidity_minted;
+
+                transfer_to_output(liquidity_minted, contract_id(), sender);
+
+                // If user sent more than the correct ratio, we deposit back the extra tokens
+                let token_extra = current_token_amount - token_amount;
+                if (token_extra > 0) {
+                    transfer_to_output(token_extra, ~ContractId::from(TOKEN_ID), sender);
+                }
+
+                minted = liquidity_minted;
             } else {
-              transfer_to_output(current_token_amount, ~ContractId::from(TOKEN_ID), sender);
-              transfer_to_output(current_eth_amount, ~ContractId::from(ETH_ID), sender);
-              minted = 0;
+                transfer_to_output(current_token_amount, ~ContractId::from(TOKEN_ID), sender);
+                transfer_to_output(current_eth_amount, ~ContractId::from(ETH_ID), sender);
+                minted = 0;
             }
-
         } else {
             assert(current_eth_amount > MINIMUM_LIQUIDITY);
 
             let initial_liquidity = current_eth_amount;
 
-            // Add fund to the reservers
-            add_reserve(TOKEN_ID, current_token_amount);
-            add_reserve(ETH_ID, current_eth_amount);
+            // Add fund to the reserves
+            storage.token_reserve = storage.token_reserve + current_token_amount;
```

Comment:
Use functions are not a good approach? The idea was to avoid `+ and -` operations between logics. The idea before was to safe and treat operator in a single place make it safer for the developer to know if is adding or removing reserves by use the `add_reserve` or `remove_reserve`

Reply:
Yeah we can keep the logic in a function.